### ISSUE TITLE
[bug] Publishing through AedesServer works, but messageId is always undefined

### ISSUE THREAD

**Issue by jemd93** [moscajs/aedes#667]
**System Information**
- Aedes: 0.46.1
- websocket-stream: 5.5.2
- aedes-persistence-redis 8.0.1
- mqemitter-redis: 4.1.0
- NodeJS: 14.17.5

**Describe the bug**
Currently in the process of replacing our old Mosca MQTT broker to use Aedes with Redis persistence and Redis emitter. I've managed to get everything working **except** for 2 strange behaviors:
1. When calling aedesServer.publish , the messageId of my message is always undefined, even if I add it myself to the object I'm publishing.
2. When publishing, the object being sent shows as a list of elements, 3 being undefined and 1 being the actual message I'm sending. (Not sure if this is expected behavior, but would like to know the reason for it if it is)

**See code and logs in "Additional Context" section**

**To Reproduce**
1. Create Aedes server with redis emitter and redis persistence
2. Publish message using QOS 1 with a messageId in the object being published

**Expected behavior**
For the first issue, the messageId should be the same one I'm using in my messageToClient object, for the second, ideally the object being printed in the logs should just be the packet, but this last one might just be expected behavior. 

**Additional context**
The code I'm using to publish is the following: 
`const dispatchToViaMqtt = ({ message, topic }) => {
    const nanoid = customAlphabet('1234567890', 7)
    const messageToClient = {
      cmd: 'publish',
      messageId: parseInt(nanoid()),
      payload: JSON.stringify(message),
      retained: false,
      dup: false,
      topic,
      qos: 1,
    };
    aedesServer.publish(messageToClient, (err, sentMsg) => {
      console.log('message sent to client', sentMsg);
    });
  };`

The log produced from the callback function:
`    message sent to client [ undefined,
      Packet {
        cmd: 'publish',
        brokerId: '1234',
        brokerCounter: 182,
        topic:
         'topic/message/id1/id2',
        payload:
         '"message goes here"',
        qos: 1,
        retain: false,
        dup: false,
        messageId: undefined },
      undefined,
      undefined ]`

**Comment by robertsLando** [comment]
@jemd93 messageId is something used by Aedes internally to have a unique identifier of a packet, you should not set it. 

> hat I understand is that the 4 functions in publishFuncsQoS are all called in succession, is this the reason for the list of 4 elements with "undefined" in 3 of them?

Based on qos level the broker `_series` will process all the functions in the array passed in serie, once finished will call `done`. I don't understand what you mean with `is this the reason for the list of 4 elements with "undefined" in 3 of them`

**Comment by jemd93** [comment]
> @jemd93 messageId is something used by Aedes internally to have a unique identifier of a packet, you should not set it.
> 

Hm, right, but even if I dont set a value for it, the value printed through my callback function is still "undefined". The code I showed in the "Additional context" section, shows how I'm calling publish including it's callback, followed by what I see in the logs when printing that message that was sent, notice in said logs, it seems to print a list of [undefined, Packet, undefined, undefined]. If I do the same thing using Mosca, the same print will only print the Packet, and the Packet will also have a messageId with an actual value. I would think that at least the MessageId should have something in it? If i change QOS to 2 or any other value in the `messageToClient` object, those changes are reflected when priting sentMsg in the callback, except, regardless of what I do, messageId is always undefined.  

> 
> Based on qos level the broker `_series` will process all the functions in the array passed in serie, once finished will call `done`. I don't understand what you mean with `is this the reason for the list of 4 elements with "undefined" in 3 of them`

Regarding this I was just theorizing if the reason `sentMsg` prints a list of [undefined, Packet, undefined, undefined] might be related to this series of 4 functions being called. Regardless of the reason, like I mentioned above, if I use Mosca instead of Aedes in my code, the `sentMsg` prints only the Packet and said Packet contains a messageId.

**Comment by robertsLando** [comment]
The second argument of fastseries is an array with the results of each function in the series, in this case if you check the code that is the return value of `enqueueOffline`, more specifically the return value of `enqueuer.done`: https://github.com/moscajs/aedes/blob/main/aedes.js#L221

In clients when a publish with qos > 0 is received we use [qos-packet](https://github.com/moscajs/aedes/blob/main/lib/qos-packet.js) that will set the messageId, here we always use `new Packet`, I think that this could be an error or I'm missing something...

@gnought @mcollina is there a specific reason why we always use new Packet [here](https://github.com/moscajs/aedes/blob/main/aedes.js#L254)?

**Comment by jemd93** [comment]
@robertsLando Yeah, I thought it might be an error, but not sure. Mostly it seems weird to me that when creating a new Packet, "undefined" is assigned to the Id and it doesn't seem to get changed anywhere else. Comparing it to the Mosca publish code https://github.com/moscajs/mosca/blob/master/lib/server.js#L344-L400 messageId is generated through `nanoid` 🤔 not sure if that should be the case here too. 

Also thank you so much for answering my questions, I really appreciate it!

**Comment by robertsLando** [comment]
In aedes we use a `_nextId` property set on client and it is incremented on each packet, it's also used to dedupe packets. We could add that property to Aedes itself and increment it like we do in qos-packet. Are you interested to submit a PR? Remeber to also add a test that covers this
